build_runner/test/integration_tests/optional_outputs_test.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build_test/build_test.dart';
+import 'package:meta/meta.dart';
+import 'package:test/test.dart';
+import 'package:test_descriptor/test_descriptor.dart' as d;
+
+import 'utils/build_descriptor.dart';
+
+// test-package-start
```

Comment:
Not a huge fan of these special comments as opposed to just a multiline string. Especially because they don't really stand out much.

Reply:
I originally had it grab everything between the last import and `main()` but that was even more magic. I wanted to make this change because as opposed to a multi line string this approach gives us analysis, refactoring, formatting, etc.

I found myself editing things in this file then cutting and pasting into the string and I thought it would be nice to automate that step and get the benefits of analysis errors when there are breaking changes.

Let's find some time to chat about this today.

Reply:
> I found myself editing things in this file then cutting and pasting into the string and I thought it would be nice to automate that step and get the benefits of analysis errors when there are breaking changes.

Ya agreed there are benefits to not having it be completely in a string - I just don't think these existing comments are obvious enough. If you don't have pre-existing knowledge about what they are doing or how this test works then this would be pretty confusing I think. Happy to chat later today :D.

Reply:
Discussed offline. We decided that there are enough benefits with this approach to make the weirdness of snipping out a chunk of code from the file worth it. It doesn't make any difference to the code which does the snipping, but I added a bunch of `####` to the comments to make them stand out more and make it clear that the chunk isn't normal code. We can continue to do that by convention.